### ISSUE TITLE
mermaid diagrams are not entirely working

### ISSUE THREAD

**Issue by f8ttyc8t** [careteditor/caret#502]
I'm using mermaid to add graphs to my documents quite often.
So I was happy to see mermaid support became part of caret.io

The bad: caret seems not to accept all mermaid graph types.

I've taken sample diagrams from mermaids page to test carets abilities.
I am using latest caret.io, tested on Windows 10 + MacOS 10.12.6, using default preview.
In my case results are:

- [x]  _graph_: works
- [x]  _sequenceDiagram_: works
- [ ]  _classDiagram_: fails
- [ ]  _gitGraph_: partly fails
- [x]  _gantt_: works

Please find attached my source (markdown) file as well as exported PDF.

[caret-mermaid-examples.md.txt](https://github.com/careteditor/caret/files/1230379/caret-mermaid-examples.md.txt)
[caret-mermaid-examples.pdf](https://github.com/careteditor/caret/files/1230380/caret-mermaid-examples.pdf)

Thanks in advance!

**Comment by f8ttyc8t** [comment]
Hi,
Any idea about when this bug will be fixed?

Thanks in advance!

**Comment by astoilkov** [comment]
Hi @f8ttyc8t,

We have investigated these problems and they are related to the poor Mermaid implementation. The library has bugs that haven't been fixed for a long time. This is why we have put Mermaid support on hold in Caret. We are also considering removing support for Mermaid after measuring the amount of people using this feature.

**Comment by f8ttyc8t** [comment]
Hi @astoilkov ,
thanks for response.
Even I feel sorrow about possibility to loose that feature I understand your arguments. 
Probably better to have a less complex but rock solid (and beatifully crafted!) tool.
(Don't know which library you are using, https://mermaidjs.github.io/ works fine for me).

Keep evolving that gem!

**Comment by fwouts** [comment]
There may be other reasons why people don't use Mermaid in Caret:
- the feature is hidden deep in menus and completely undocumented
- the implementation bugs are bad enough that serious Mermaid users will stop using Caret (I've personally had to switch over to VS Code)

If you fixed the implementation and helped people discover and use it, you'd probably see significant uptake on it. Good luck, whichever way you go!

**Comment by f8ttyc8t** [comment]
@astoilkov - I totally agree to @fwouts.
Built in diagram support is an tremendous asset for all kind of writers. Similar to LaTeX.

Anyway - hope Caret continues supporting Mermaid (or something like that)!

**Comment by repli-a** [comment]
I paid $30 for this and have had to switch to VSCode because Mermaid doesn't work. It's a real shame. +1 for this being sorted pronto.

**Comment by astoilkov** [comment]
Hi @bisongde,

What are the problems experienced with Mermaid? `classDiagram` usage, `gitGraph` or something else?

Thanks in advance.

P.S. We will take special attention to Mermaid soon. Thanks again.

**Comment by RasmusKjeldmand** [comment]
how is it going with the Mermaid plugin? I would like it since

**Comment by rasmusmk** [comment]
When using classDiagram nothing shows up in preview. ex.
```mermaid
classDiagram
  Class01 <|-- AveryLongClass : Cool
  Class03 *-- Class04
  Class05 o-- Class06
  Class07 .. Class08
  Class09 --> C2 : Where am i?
  Class09 --* C3
  Class09 --|> Class07
  Class07 : equals()
  Class07 : Object[] elementData
  Class01 : size()
  Class01 : int chimp
  Class01 : int gorilla
  Class08 <--> C2: Cool label
```
Which version of mermaid is used?

**Comment by f8ttyc8t** [comment]
Did you think about using Graphviz instead of Mermaid (e.g. https://github.com/mdaines/viz.js/)?
Because I can't see any progress related to these issues.

**Comment by astoilkov** [comment]
Yes. Graphviz is a possibility but it is more likely we fix all mermaid issues. Regarding the progress on these issues - we have some other more priority tasks before we move to these ones.

**Comment by f8ttyc8t** [comment]
@astoilkov Thank you for response.

Looking forward testing next beta - v4 is already pretty good!

**Comment by f8ttyc8t** [comment]
Maybe I've missed something, but mermaid still does not work.
Released version of Caret (3.4.6) still facing issuses with given test files (first article in this section).
Beta version of Caret (4.0.0-rc13) does not show any diagram at all in preview.

So, at least for me, this is still an open issue (not resolved).

**Comment by repli-a** [comment]
@f8ttyc8t I think the resolution was "give up."

**Comment by astoilkov** [comment]
Hi @f8ttyc8t,

Unfortunately, we decided to remove mermaid diagrams from Caret for now. It seems they would have been a difficult task to improve/fix. We have features that a lot more users are using and we want to focus on them. We are a small team and we are very careful when deciding on what things we should focus on in order to be successful.

I hope you can understand our viewpoint.

**Comment by f8ttyc8t** [comment]
@astoilkov even I am unhappy with your decision - thank you for being honest.
